ouroboros-network/src/Ouroboros/Network/Socket.hs

```diff
@@ -157,10 +163,11 @@ withNetworkNode NetworkInterface {nodeAddress, protocols} k =
     -- Make the server listening socket
     mkSocket :: IO Socket.Socket
     mkSocket = do
-      sd <- Socket.socket Socket.AF_INET Socket.Stream Socket.defaultProtocol
-      Socket.setSocketOption sd Socket.ReuseAddr 1
+      sd <- Socket.socket (Socket.addrFamily nodeAddress) Socket.Stream Socket.defaultProtocol
+      when (Socket.addrFamily nodeAddress /= Socket.AF_UNIX) $ do
```

Comment:
I would prefer a more explicit:
```
case Socket.addrFamily nodeAddress of
  addr | addr == Socket.AF_INET || addr == Socket.AF_INET6 -> do ...
  _   -> return ()
```
But if it's unlikely that we're use any other `AddressFamily` than inet/inet6/unix then it's probably fine.

Reply:
We only test and support inet/inet6/unix, but this patch inadvertently enables support for _all_ families, and there are a lot, http://hackage.haskell.org/package/network-3.1.0.0/docs/Network-Socket.html#t:Family . How about doing an explicit test at the start of `withNetworkNode` and fail if it is an unsupported family?

Reply:
That's ok; I still prefer `==` over `/=` just for readability.

Reply:
Will address.